### DISCUSSION TITLE
Are these memory leak findings valid?

### DISCUSSION THREAD

**Discussion by farkasdvd** [nlohmann/json#4031] [Q&A] [answered]
Hello,
Our static analysis tool (Coverity) has detected a couple of memory leaks in the `json_value` union. Here is an example (there is a same finding for every ctor which calls `create()`):

![image](https://github.com/nlohmann/json/assets/30418389/c1ecdf4b-bfba-4d5a-b28c-ca9d99436f3b)

Could someone who is familiar with the design confirm the validity of these findings? In the `json_value` union there is a `destroy()` method, which I assume is explicitly called before a `json_value` object goes out of scope. Is that the case?

**Comment by gregmarr** [comment, marked as answer]
Yes, the `json_value` object is contained in `struct data` and `json_value::destroy()` is called in `data`'s destructor here: 

https://github.com/nlohmann/json/blob/develop/include/nlohmann/json.hpp#L4223

        ~data() noexcept
        {
            m_value.destroy(m_type);
        }

The `unique_ptr` is used in the `create` function so that the memory is released if the `T` constructor throws.
